ludwig/data/preprocessing.py

```diff
@@ -1293,11 +1293,18 @@ def build_preprocessing_parameters(
             if TYPE in feature_config[ENCODER]:
                 encoder_class = get_encoder_cls(feature_config[TYPE], feature_config[ENCODER][TYPE])
                 if hasattr(encoder_class, "fixed_preprocessing_parameters"):
+                    # Update preprocessing parameters for text encoders using fixed_preprocessing_parameters
                     encoder_fpp = encoder_class.fixed_preprocessing_parameters
 
                     preprocessing_parameters = merge_dict(
                         preprocessing_parameters, resolve_pointers(encoder_fpp, feature_config, "feature.")
                     )  # TODO(Connor): Temporary fix, refactor this during preproc refactor
+                elif callable(getattr(encoder_class, "get_fixed_preprocessing_params", None)):
```

Comment:
Instead of doing this, we should use good object-oriented principles and have a function called `get_fixed_preprocessing_params` that exists in the base Encoder class. This can return `{}`. Then we override this function in subclasses.

Similarly, this new function would subsume the previous `fixed_preprocessing_parameters` attribute above, so the code would be simplified to:

```
preprocessing_parameters = merge_dict(
                        preprocessing_parameters, encoder_class.get_fixed_preprocessing_params(feature_config[ENCODER])
                    )
```

And we can remove these two `if-else` blocks. Make sense?

Reply:
@tgaddair Totally agree, thanks for pointing this out. This is the cleanest way forward! Will update